src/velocitas/model_generator/tree_generator/file_import.py

```diff
@@ -31,30 +31,51 @@ def __str__(self):
 
 
 class FileImport:
+
     def __init__(
-        self, file_path: str, include_dirs: List[str], strict: bool, overlays: List[str]
+        self,
+        file_path: str,
+        unit_file_path_list: str,
+        include_dirs: List[str],
+        strict: bool,
+        overlays: List[str],
     ):
         self.file_path = file_path
         self.include_dirs = include_dirs
         self.strict = strict
         self.overlays = overlays
         # setting the file format implementation object from the file_path
-        self.format_implementation = self.__get_format_implementation(self.file_path)
+        self.format_implementation = self.__get_format_implementation(
+            self.file_path, unit_file_path_list
+        )
 
-    def __get_format_implementation(self, file_path: str):
+    def __get_format_implementation(
+        self,
+        file_path: str,
+        unit_file_path_list: List[str],
+    ):
         file_ext = os.path.splitext(file_path)[1][1:]
+        for unit_file_path in unit_file_path_list:
```

Comment:
Introduce function to verify the extensions of the unit file paths.

Reply:
its two lines under or you want me to split that in a separate function? since its just three lines of code I kept it in place.